dockhub.py

```diff
@@ -1,226 +1,182 @@
 #!/usr/bin/env python3
 
+from functools import wraps
 import json
 from os import getenv
-from sys import exit
+import sys
 
 import click
 import requests
 
 
-def die(msg=None, exception=None):
-    if msg is not None:
-        print(f'{msg}')
-    if exception is not None:
-        print(f'{exception}')
-    exit(1)
+CONTENT_HEADER = {'Content-Type': 'application/json', 'charset': 'utf-8'}
+DOCKERHUB_URL = 'https://hub.docker.com/v2'
 
 
+def handle_http_errors(fun):
+    """Wrap function with exception handling for common HTTP errors."""
+    @wraps(fun)
```

Comment:
The wrapped function `_handle_http_errors` is returned by this function. Because of that, it's got all the wrong function metadata, so we wrap it and copy the metadata from `fun` into `_handle_http_errors`. The metadata includes debugging information like the name of the function, which file it's in, what line, etc.